### PR TITLE
Restore cdc log failed due to excess version part in ddl file when using s3 sink

### PR DESCRIPTION
**Pull request by yuqi1129**
### What problem does this PR solve? 
see https://github.com/pingcap/ticdc/issues/1422

### What is changed and how it works?

only when `firstCreated` is true and previous log size is 0 will write `version` to ddl cdc log

### Check List 

Tests 

 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### CODE REVIEW

File: cdc/sink/cdclog/s3.go [hunk 1]
```diff
@@ -289,6 +284,26 @@ func (s *s3Sink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
 	if err != nil {
 		return cerror.WrapError(cerror.ErrS3SinkStorageAPI, err)
 	}
+
+	// only reboot and size = 0 should we add version to s3
+	withVersion := firstCreated && size == 0
```
Comment (amyangfei) [inline cdc/sink/cdclog/s3.go:289]: If `size > maxDDLFlushSize`, should the `withVersion` be true?

Comment (yuqi1129) [reply cdc/sink/cdclog/s3.go:289]: Oh, you are right, maxDDLFlushSize should also be considered

**Review by amyangfei — COMMENTED** [review]
LGTM

**Comment by amyangfei** [comment]
PTAL @leoppro @liuzix

**Comment by liuzix** [comment]
@3pointer PTAL. BTW is it possible to add some tests for this case?

**Comment by yuqi1129** [comment]
> @3pointer PTAL. BTW is it possible to add some tests for this case?

What's your opinion?

**Comment by yuqi1129** [comment]
Similar test case, please see PR https://github.com/pingcap/ticdc/pull/1515/files

**Comment by overvenus** [comment]
LGTM, thanks!

**Comment by yuqi1129** [comment]
Can anyone help to merge this PR if there is not problem~

**Pull request merged**